tests/virt_autotest/virt_utils.pm

```diff
@@ -617,4 +625,56 @@ sub perform_guest_restart {
     }
 }
 
+
+#return 1 if it is a fv guest judging by name
+#feel free to extend to support more cases
+sub is_fv_guest {
+    my $guest = shift;
+    return $guest =~ /\bfv\b/ || $guest =~ /\bhvm\b/;
+}
+
+#return 1 if it is a pv guest judging by name
+#feel free to extend to support more cases
+sub is_pv_guest {
+    my $guest = shift;
+    return $guest =~ /\bpv\b/;
+}
+
+#return 1 if test is expected to run on KVM hypervisor
+sub is_kvm_host {
+    return check_var("SYSTEM_ROLE", "kvm") || check_var("HOST_HYPERVISOR", "kvm");
+}
+
+#return 1 if test is expected to run on XEN hypervisor
+sub is_xen_host {
+    return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen");
+}
+
+#check host to make sure it works well
+#return 'OK' when the host works well, or other string to flag the state of the host.
+#welcome everybody to extend this function
+sub check_host {
+    return "OK";
+}
```

Comment:
What's the purpose of this?

Reply:
just list a empty function here. anybody including me will extend it in the future test when we get more detailed failure or what checkpoint we should do to the host and guest.